### PR TITLE
Add specific export value function for select driver

### PR DESCRIPTION
**Pull request by alicecd**

### CODE REVIEW

**Comment by shaoshiva** [comment]
You should use the trait `Trait_Driver_Field_Choices_Single` as for the `Driver_Field_Radio`.

**Comment by alicecd** [comment]
The renderExportValue from Trait_Driver_Choices_Single doesn't print anything in the exported file for Select fields.

**Comment by shaoshiva** [comment]
You should better fix the `renderExportValue` in `Trait_Driver_Field_Choices_Single` instead of re-implementing theses methods. The code should be the same as `renderAnswerHtml` but without the `e()` on the returned value.

Btw the radio driver may also contain this bug, so fixing the trait will fix this driver too.

File: classes/trait/driver/field/choices/single.php [hunk 1]
```diff
@@ -104,7 +91,18 @@ protected function getValueChoiceLabel($value)
...
 
-        return \Arr::get($choices, $value, $value);
+        $selectedValue = '';
+
+        $hashValue = $this->convertChoiceValueToHash($value);
+        if (array_key_exists($hashValue, $choices)) {
+            $selectedValue = \Arr::get($choices, $hashValue);
+        }
+
+        if (empty($selectedValue)) {
```
Comment (jguyomard) [inline classes/trait/driver/field/choices/single.php]: can be replaced by a `else` instruction

Comment (alicecd) [reply classes/trait/driver/field/choices/single.php]: Done

File: classes/trait/driver/field/choices/single.php [hunk 2]
```diff
@@ -20,7 +20,6 @@ public function renderAnswerHtml(Model_Answer_Field $answerField)
         // Gets the answer value
         $value = $this->sanitizeValue($answerField->value);
 
-        // Converts to choice label
```
Comment (Meroje) [inline classes/trait/driver/field/choices/single.php:23]: Should keep the comment

File: classes/trait/driver/field/choices/single.php [hunk 3]
```diff
@@ -37,13 +36,7 @@ public function renderExportValue(Model_Answer_Field $answerField)
         // Gets the answer value
         $selectedValue = $this->sanitizeValue($answerField->value);
 
-        // Gets the choices list
```
Comment (Meroje) [inline classes/trait/driver/field/choices/single.php:40]: Should keep the comment

File: classes/trait/driver/field/choices/single.php [hunk 4]
```diff
@@ -79,17 +72,11 @@ public function getEmailData($inputValue, Model_Answer $answer)
         // Gets the answer value
         $selectedValue = $this->sanitizeValue($inputValue);
 
-        // Gets the choices list
```
Comment (Meroje) [inline classes/trait/driver/field/choices/single.php:82]: Should keep the comment

File: classes/trait/driver/field/choices/single.php [hunk 5]
```diff
@@ -37,13 +36,7 @@ public function renderExportValue(Model_Answer_Field $answerField)
...
-        $choices = $this->getChoicesList();
-
-        $export    = '';
-        if( array_key_exists($selectedValue, $choices) ){
-            $export = e($choices[$selectedValue]);
-        }
+        $export = $this->getValueChoiceLabel($selectedValue);
 
         return $export;
```
Comment (Meroje) [inline classes/trait/driver/field/choices/single.php]: You removed the `e()`, maybe check if you should keep it here ?

File: classes/trait/driver/field/choices/single.php [hunk 6]
```diff
@@ -104,7 +91,16 @@ protected function getValueChoiceLabel($value)
         // Gets the choices
         $choices = $this->getChoicesList();
 
-        return \Arr::get($choices, $value, $value);
+        $selectedValue = '';
```
Comment (Meroje) [inline classes/trait/driver/field/choices/single.php]: Don't really need to initialize this since you have an `else` below

**Review by Meroje — CHANGES_REQUESTED** [review]

File: classes/trait/driver/field/choices/single.php (hunk 4 above)
Comment (jguyomard) [reply classes/trait/driver/field/choices/single.php:82]: nop, `getValueChoiceLabel()` doesn't return the choices list...

If you want a comment here, `Gets the answer label` or `Converts to choice label` is better

File: classes/trait/driver/field/choices/single.php (hunk 5 above)
Comment (shaoshiva) [reply classes/trait/driver/field/choices/single.php]: It's a value for a CSV file and it's already escaped outside the driver (when generating the CSV file with the values) :
```php
\Format::forge($csv)->to_csv()
```

File: classes/trait/driver/field/choices/single.php (hunk 2 above)
Comment (shaoshiva) [reply classes/trait/driver/field/choices/single.php:23]: :+1:

File: classes/trait/driver/field/choices/single.php (hunk 3 above)
Comment (shaoshiva) [reply classes/trait/driver/field/choices/single.php:40]: :+1:

**Review by jguyomard — APPROVED** [review]

File: classes/trait/driver/field/choices/single.php (hunk 4 above)
Comment (shaoshiva) [reply classes/trait/driver/field/choices/single.php:82]: :+1:

Comment (alicecd) [reply classes/trait/driver/field/choices/single.php:82]: Done for all

**Review by shaoshiva — APPROVED** [review]

**Review by Meroje — APPROVED** [review]

**Pull request merged**